robots/pr-labeler/main.go

```diff
@@ -0,0 +1,149 @@
+limitations under the License.
+*/
+
+// Commenter provides a way to --query for issues and append a --comment to matches.
+//
+// The --token determines who interacts with github.
+// By default commenter runs in dry mode, add --confirm to make it leave comments.
+// The --updated, --include-closed, --ceiling options provide minor safeguards
+// around leaving excessive comments.
+package main
+
+import (
+	"flag"
+	"log"
+	"net/url"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/github"
+)
+
+const (
+	needsOkToTest = "needs-ok-to-test"
+	okToTest      = "ok-to-test"
+)
+
+type client interface {
+	AddLabel(org, repo string, number int, label string) error
+	GetIssueLabels(org, repo string, number int) ([]github.Label, error)
+	GetPullRequests(org, repo string) ([]github.PullRequest, error)
+	ListCollaborators(org, repo string) ([]github.User, error)
+	ListOrgMembers(org, role string) ([]github.TeamMember, error)
+}
+
+type options struct {
+	confirm  bool
+	endpoint flagutil.Strings
+	org      string
+	repo     string
+	token    string
+}
+
+func flagOptions() options {
+	o := options{
+		endpoint: flagutil.NewStrings("https://api.github.com"),
+		org:      "kubernetes",
+		repo:     "test-infra",
+	}
+	flag.BoolVar(&o.confirm, "confirm", false, "Mutate github if set")
+	flag.StringVar(&o.token, "token", "", "Path to github token")
+	flag.Parse()
+	return o
+}
+
+func main() {
+	log.SetFlags(log.LstdFlags | log.Lshortfile)
+	o := flagOptions()
+
+	if o.token == "" {
+		log.Fatal("empty --token")
+	}
+
+	secretAgent := &config.SecretAgent{}
+	if err := secretAgent.Start([]string{o.token}); err != nil {
+		log.Fatalf("Error starting secrets agent: %v", err)
+	}
+
+	var err error
+	for _, ep := range o.endpoint.Strings() {
+		_, err = url.ParseRequestURI(ep)
+		if err != nil {
+			log.Fatalf("Invalid --endpoint URL %q: %v.", ep, err)
+		}
+	}
+
+	var c client
+	if o.confirm {
+		c = github.NewClient(secretAgent.GetTokenGenerator(o.token), o.endpoint.Strings()...)
+	} else {
+		c = github.NewDryRunClient(secretAgent.GetTokenGenerator(o.token), o.endpoint.Strings()...)
+	}
+
+	// get all open PRs
+	prs, err := c.GetPullRequests(o.org, o.repo)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// get the list of members once and use a set for lookups
+	mem, err := c.ListOrgMembers(o.org, "member")
+	if err != nil {
+		log.Fatal(err)
+	}
+	members := sets.NewString()
+	for _, member := range mem {
+		members.Insert(member.Login)
+	}
+
+	// get the list of collaborators once and use a set for lookups
+	col, err := c.ListCollaborators(o.org, o.repo)
+	if err != nil {
+		log.Fatal(err)
+	}
+	collaborators := sets.NewString()
+	for _, collaborator := range col {
+		collaborators.Insert(collaborator.Login)
+	}
+
+	for _, pr := range prs {
+		// skip PRs from members
+		if members.Has(pr.User.Login) {
+			continue
+		}
+		// skip PRs from collaborators
```

Comment:
This should be a flag.

While by default prow trusts repo collaborators, we do not trust collaborators in the kubernetes org: https://github.com/kubernetes/test-infra/blob/2264543fd4ad973cfb838a9b4a17cf37dea81f76/prow/plugins/plugins.go#L395-L397